### DISCUSSION TITLE
3.0 migration, docs questions

### DISCUSSION THREAD

**Discussion by MrAwesome** [kheina-com/Blue-Blocker#123] [General]
@DanielleMiu - I'd love to help you out getting 3.0 out the door, have a few questions for whenever you have the time and energy:

1) You mentioned that there are some 2.6 changes that need to be ported to 3.0, do you have a list of those handy?
2) Are there any other changes or code migrations that need to happen before 3.0 is ready?
3) Are there any documentation changes that need to happen other than the README changes already there?
4) Does anything need to change about the deploy process for 3.0? (Also, if you have a sec to write it up I'd love to know what the process is, never done anything with browser extensions until now)

**Comment by kheina** [comment]
1. pretty much everything in the [0.2.6 llb/diff](https://github.com/kheina-com/Blue-Blocker/pull/99). iirc @jaredkotoff made the ts branch from 0.2.5 and I meant to update it myself and get the hang of ts at the same time but it just never happened
2. I don't believe so, outside your changes the other day
3. I'm not sure actually. I haven't had the chance to look through the ts code with as much detail as I would like yet. I know it works and when I gave it the once-over it looked good but that's really all I've done
4. @jaredkotoff is probably the better person to ask about the ts procedure. all I do now is run the `make chrome` and `make firefox` commands in the make file, check to make sure they're working, and then upload them to the chrome website and firefox addons. knowing vite, it might need to go through `npm run build`, but I think he actually got that wrapped into the makefile too. at some point I'd like to make a github action to do the make/upload part automatically but it's probably best to wait until firefox can be updated to [manifest v3](https://github.com/kheina-com/Blue-Blocker/issues/39) so it's not building it twice

**Reply by jaredcat** [reply]
The Readme should cover everything but to clarify:
Just run `npm run dev` and load into chrome. It`ll hotreload as you go. 
NOTE: styling on the extension window doesn't seem to load here but does when built. Couldn't figure out why. 

Make file for final builds (chrome can use `npm run build` now though), it (hack-ly) handles v2 for Firefox. I also left [this comment](https://github.com/guocaoyi/create-chrome-ext/discussions/33#discussion-5192359) in case there's a better solution I'm missing
I think once FF does v3 we won't need the makefile.

With that said I don't think any contributers really need to worry about these builds besides for final testing since none of us can submit them.

**Reply by jaredcat** [reply]
@MrAwesome @DanielleMiu I rebased v0.3.0 off of main: https://github.com/kheina-com/Blue-Blocker/pull/126
If this looks good, I think we can probably swap current `llb-v0.3.0` branch for this one?

**Reply by kheina** [reply]
yeah, probably. in the end it doesn't really matter, we'll just close whichever pr doesn't get merged. I'm gonna look more at it today and do more testing on your branch

**Reply by MrAwesome** [reply]
For 0.3.0, is there anything I can help with, or will I just need to wait for #126?

**Reply by kheina** [reply]
testing and bug fixing and making sure all changes from 0.2.6 are in. I was looking at code earlier and saw it's still using xhr rather than the new fetch logic

**Reply by jaredcat** [reply]
Yea my rebase needs to be cleaned up. I was going to do it once I got sometime back from vacation. But extra eyes on it either way would be good. Notes on the PR would be helpful

**Comment by kheina** [comment]
I might want it updated to use tabs since I think it's currently using spaces, but that's all I see taking a quick second look

**Comment by kheina** [comment]
working on this lately, also adding that verification db. I'm using the checklist I posted in the rebase branch as my guide

**Comment by kheina** [comment]
#143 has all of the rebase changes. I rebased again on main to fix the manifest issue, but I guess that messed up everything in the pr. you can rebase again if we want to use that branch or we can switch (again lol) to the new new rebased branch

**Reply by kheina** [reply]
ok so I built it for chrome and firefox manually, and they seem to be working fine. there is still that issue with the queue consumer whenever it needs to wait and then seems to block a bunch at once that I need to look into, but otherwise everything looks good. 

[blue-blocker-firefox-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11858490/blue-blocker-firefox-0.3.0.zip)
[blue-blocker-chrome-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11858491/blue-blocker-chrome-0.3.0.zip)

**Reply by kheina** [reply]
these are both built off https://github.com/kheina-com/Blue-Blocker/pull/143/commits/9ee934f4e60ccb55369da89862cb52a0439db264

**Reply by kheina** [reply]
> there is still that issue with the queue consumer whenever it needs to wait and then seems to block a bunch at once that I need to look into

this looks like a bigger issue than I thought, it just burned my whole queue after queue was locked for like 5 seconds. could trigger logouts

**Reply by kheina** [reply]
fixed that issue. (finally) and another asset url issue too. these are built from https://github.com/kheina-com/Blue-Blocker/pull/143/commits/e778b75a751a0a9986f07a423f8e7371d9306496

[blue-blocker-firefox-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11858781/blue-blocker-firefox-0.3.0.zip)
[blue-blocker-chrome-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11858782/blue-blocker-chrome-0.3.0.zip)

**Reply by kheina** [reply]
I'm dumb, I've been copy and pasting the same manifest for firefox as the first, and the hashes broke everything. here's the updated one from https://github.com/kheina-com/Blue-Blocker/commit/e778b75a751a0a9986f07a423f8e7371d9306496

[blue-blocker-firefox-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11858858/blue-blocker-firefox-0.3.0.zip)

**Reply by kheina** [reply]
if I find no issues, I'll merge to main and upload these tomorrow. holy shit you guys, typescript is happening!!

**Reply by kheina** [reply]
I thought of a way to make indexed db work by using the service worker to make all accesses to the db and also fixed the external message listener for external addon integrations

built from https://github.com/kheina-com/Blue-Blocker/pull/143/commits/5f0105ac420d34dfc7692373a74b8394c371af2b
[blue-blocker-firefox-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11861141/blue-blocker-firefox-0.3.0.zip)
[blue-blocker-chrome-0.3.0.zip](https://github.com/kheina-com/Blue-Blocker/files/11861142/blue-blocker-chrome-0.3.0.zip)

**Reply by kheina** [reply]
gonna start posting these on the pr itself for simplicity
